scarlet/lite/component.py

```diff
@@ -0,0 +1,621 @@
+        return self._morph_step
+
+    def get_model(self, bbox=None, frame=None):
+        """Generate the full model for this component"""
+        model = self.sed[:, None, None] * self._morph(frame)[None, :, :]
+
+        if bbox is not None:
+            slices = overlapped_slices(bbox, self.bbox)
+            _model = np.zeros(bbox.shape, self.morph.dtype)
+            _model[slices[0]] = model[slices[1]]
+            model = _model
+        return model
+
+    def prox_sed(self, sed, prox_step=0):
+        """Apply a prox-like update to the SED
+
+        Parameters
+        ----------
+        sed: `numpy.ndarray`
+            The SED of the model.
+        prox_step: `float`
+            A scaling parameter used in some proximal operators,
+            but ignored here.
+        """
+        # prevent divergent SED
+        sed[sed < self.floor] = self.floor
+        return sed
+
+    def grad_sed(self, input_grad, sed, morph):
+        """Gradient of the SED wrt. the component model
+
+        Parameters
+        ----------
+        input_grad: `numpy.ndarray`
+            Gradient of the likelihood wrt the component model
+        sed: `numpy.ndarray`
+            The model of the SED.
+        morph: `numpy.ndarray`
+            The model of the morphology.
+
+        Returns
+        -------
+        result: `float`
+            The gradient of the likelihood wrt. the SED.
+        """
+        _grad = np.zeros(self.bbox.shape, dtype=self.sed.dtype)
+        _grad[self.slices[1]] = input_grad[self.slices[0]]
+        return np.einsum("...jk,jk", _grad, morph)
+
+    def update(self, it, input_grad):
+        """Update the component
+
+        Parameters
+        ----------
+        it: `int`
+            The current iteration of the optimizer.
+        input_grad: `numpy.ndarray`
+            Gradient of the likelihood wrt the component model
+        """
+        sed = self.sed.copy()
+        morph = self.morph
+        self._sed.update(it, input_grad, morph)
+        self._params.update(it, input_grad, self, morph, sed, self.frame)
+
+    def resize(self):
+        """Resize the box that contains the model
+
+        Not yet implemented, so for now the model box
+        does not grow. In the long run this will be
+        based on a cutoff value for the model.
+        """
+        return False
+
+    def init_adaprox(self, noise_rms, max_prox_iter=1, factor=10):
+        """Convert all of the parameters into adaprox parameters
+
+        Parameters
+        ----------
+        noise_rms: `numpy.ndarray`
+            The RMS noise in each band.
+        max_prox_iter: `int`
+            Maximum number of proximal iterations.
+        factor: `int`
+            The factor to scale the noise to set the
+            SED step.
+        """
+        self._sed = AdaproxParameter(
+            self._sed.x,
+            step=partial(relative_step, factor=1e-2, minimum=noise_rms / factor),
```

Comment:
Should the other factor arg here be constant?

Reply:
Yes... probably. Another carry over from old scarlet, but I'll make the change.